.travis.yml

```diff
@@ -3,61 +3,53 @@ rust: &rust
   sudo: required
   dist: trusty
   addons:
-      apt:
-          packages:
-              - libssl-dev
-              - libcurl4-openssl-dev
-              - libelf-dev
-              - libdw-dev
-              - cmake
-              - gcc
-              - binutils-dev
-              - libc6-dev
+    apt:
+      packages:
+        - git
+        - libssl-dev
+        - libcurl4-openssl-dev
+        - libelf-dev
+        - libdw-dev
+        - cmake
+        - gcc
+        - binutils-dev
+        - libc6-dev
   cache:
     directories:
       - /home/travis/.cargo
+
   before_cache:
     - rm -rf /home/travis/.cargo/registry
+
   env:
     - RUST_BACKTRACE=1
     - RUSTFLAGS=-Ctarget-feature=+aes,+ssse3
     - PATH=$HOME/.local/bin:$PATH
-    - LD_LIBRARY_PATH=$HOME/lib
-    - PKG_CONFIG_PATH=$HOME/lib/pkgconfig
+    - LD_LIBRARY_PATH=/home/travis/lib
+    - LIBRARY_PATH=/home/travis/lib
+    - PKG_CONFIG_PATH=/home/travis/lib/pkgconfig
   before_install: # versions from https://github.com/erickt/rust-zmq/blob/master/.travis.yml
-    - wget https://github.com/jedisct1/libsodium/releases/download/1.0.16/libsodium-1.0.16.tar.gz
-    - tar xvfz libsodium-1.0.16.tar.gz
-    - cd libsodium-1.0.16 && ./configure --prefix=$HOME && make && make install && cd ..
-    - wget https://github.com/zeromq/libzmq/releases/download/v4.2.5/zeromq-4.2.5.tar.gz
-    - tar xvfz zeromq-4.2.5.tar.gz
-    - cd zeromq-4.2.5 && ./configure --prefix=$HOME --with-libsodium && make && make install && cd ..
+    - git clone https://github.com/leejw51crypto/ChainTravis.git /home/travis/tmpdir
+    - cp -Rf /home/travis/tmpdir/* /home/travis
```

Comment:
travis has this cache configuration:

```
  cache:
     directories:
       - /home/travis/.cargo
```

instead of cloning a custom repo with binaries, it may be better just to extend this cache configuration -- the preparation script / before_install would check whether the binary is in the cache; if not, it'll build it (as before)... if the CI is successful, it'll save these binaries / make sure they are cached

Reply:
ok